### PR TITLE
updating microsoft teams webhook detector to use tri-state verification

### PR DESCRIPTION
**Pull request by 0x1**
### Description:
updating microsoft teams webhook detector to use tri-state verification

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go [hunk 1]
```diff
@@ -48,19 +51,33 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			Raw:          []byte(resMatch),
 		}
 		if verify {
+			client := s.client
```
Comment (ahrav) [inline pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go:54]: Suggestion: All the logic looks great to me. The nested if/else is definitely a little hard to follow. Do you think we could do something like the following to reduce the nesting and maybe even make the flow a little more clear:
```
for _, match := range matches {
...
		if verify {
			client := s.client
			if client == nil {
				client = defaultClient
			}

			isVerified, verificationErr := verifyWebhook(ctx, client, resMatch)
			s1.Verified = isVerified
			s1.VerificationError = verificationErr
		}

		if !s1.Verified && detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, false) {
			continue
		}

		results = append(results, s1)
...
}

func verifyWebhook(ctx context.Context, client *http.Client, webhookURL string) (bool, error) {
	payload := strings.NewReader(`{'text':''}`)
	req, err := http.NewRequestWithContext(ctx, "POST", webhookURL, payload)
	if err != nil {
		return false, err
	}
	req.Header.Add("Content-Type", "application/json")

	res, err := client.Do(req)
	if err != nil {
		return false, err
	}
	defer res.Body.Close()

	body, err := io.ReadAll(res.Body)
	if err != nil {
		return false, err
	}

	switch {
	case res.StatusCode == http.StatusBadRequest:
		if strings.Contains(string(body), "Text is required") {
			return true, nil
		}
		return false, fmt.Errorf("unexpected response body: %s", string(body))
	case res.StatusCode < 200 || res.StatusCode >= 500:
		return false, fmt.Errorf("unexpected HTTP response status: %d", res.StatusCode)
	default:
		return false, nil
	}
}
```

**Review by ahrav — APPROVED** [review]

File: pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go (hunk 1 above)
Comment (0x1) [reply pkg/detectors/microsoftteamswebhook/microsoftteamswebhook.go:54]: love it, that's way cleaner!
